src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,47 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  box-sizing: border-box;
+  gap: $gap;
+  max-width: $cell-size * 7 + $gap * 6;
+  margin: 80px auto;
+  line-height: $cell-size;
+  text-align: center;
+
+  @each $day, $number in $week-days {
+    &--first-day--#{$day}>.calendar__day:first-child {
+      margin-left: ($cell-size + $gap) * $number;
+    }
+  }
+
+  &__day {
+    display: flex;
+    box-sizing: border-box;
+    height: $cell-size;
+    width: $cell-size;
+    background-color: #eee;
+    border: 1px solid #000;
+    justify-content: center;
+    align-items: center;
+    transition: all 0.3s;
```

Comment:
do not add `transition` for `all` properties, it is bad performance-wise. Specify only the properties you need.